### DISCUSSION TITLE
Changing The Logo

### DISCUSSION THREAD

**Discussion by getqas** [searxng/searxng#2032] [General]
I figured out how to change the logos. I have seen lots of interest in this matter on the web but none of the answers were making sense. I am very new to python, docker and the like. I only found it through network chuck youtube video.

I fell in love with the application but wanted to change the logos to conform to my requirements.

As a PHP developer, this was all new to me. I documented each step in the process.

This is how you do it. 

First, you need to get a fork of the original searxng/searxng and a fork of searxng/searxng-docker

Then make your changes to the fork of searxng/searxng. In my case, I just wanted to change the logo images. Any changes to the application must be done in the original fork not the docker one as it is read only.

The image files are here : https://github.com/searxng/searxng/tree/master/searx/static/themes/simple/img

I deleted the existing svg and png and replaced with mine.  I named the new images exactly as the ones that were deleted.

Next you clone the forked/altered searxng/searxng repository to your machine.

Then you build an image of it (you must have docker installed on your machine)

Docker creates read only versions of the application that run in containers.

Then you need to upload your docker image to your online registry with docker hub. You need a free account with docker https://www.docker.com/.

Then you make changes to your fork of of searxng/searxng-docker/docker-compose.yaml

You have to point the docker-compose.yaml file to your docker hub registry and image version. You change code in the searxng section.

You change line 38 and 39 of that file to :
 container_name: yourusername
 image: yourusername/my-repo:latest

Then use your fork to complete the installation to Linode or wherever you are installing it and it will work.

The docker commands to perform the build of the image and push to your registry are :

build the image:
`docker build -t username/my-repo .`
(don't forget the space and the period)

rename the image: ( you need the name of the image) - it also shows up in docker desktop.
`docker tag cblah518a41981a6a username/my-repo`

push to your docker registry:
`docker push username/my-repo:latest`

**Comment by return42** [comment]
> I have seen lots of interest in this matter on the web but none of the answers were making sense.

That's true, all are very hackish IMO

> I fell in love with the application but wanted to change the logos to conform to my requirements.

I haven't thought it to its ends but if it is only the logo and the favicon I think it might be good to have options in the `/etc/searxng/settings.yml` pointing to files to used for them.

If this is where the majority looks for, it might be time to add a [feature request](https://github.com/searxng/searxng/issues/new?assignees=&labels=enhancement&template=feature-request.md&title=Add+options+for+logo+and+favicon).

**Reply by getqas** [reply]
I would absolutely agree that this should be something that doesn't require a fork and is a sought after feature. It could lead to wider acceptance too among developers and users.

For me, it was a great learning experience  to get my feet wet with docker which is a really amazing technology.

**Comment by WavedirectTel** [comment]
Agree - feature request.

**Comment by MatthK** [comment]
I would support that feature request as well. However, instead of having to fork and build they whole container, I think you can also just copy the two logo files into the existing container, to avoid all the hassle. You will have to do that once every time you update the container, but it is quite simple. Simply copy the two files as follows:

First you have to copy the two files to your searxng folder (where the settings.yml is) and then open a shell inside the container and copy the two files.

```
docker exec -it container sh
cp /etc/searxng/searxng.png /usr/local/searxng/searx/static/themes/simple/img/
cp /etc/searxng/searxng.svg /usr/local/searxng/searx/static/themes/simple/img/
```
